### PR TITLE
EXM 104 R4 Support

### PR DESCRIPTION
**Pull request by okeefm**
Support for EXM 104 R4 in `fhir-patient-generator`

Also, migrated to using `connectathon`'s master branch, and using resources out of there instead of out of this repo, to mirror luke's PR.

### Testing instructions
* `make clean && make r4`
* Also probably worth testing `make clean && make` to test the STU3 patient generation

Before this can be merged

 After projecttacoma/synthea#10 is merged, update branch in Makefile

### CODE REVIEW

**Review by mgramigna — CHANGES_REQUESTED** [review]
Looks good. One suggestion that I have is to not include the `fhir` and `fhir_stu3` directories in the `patients-r4` and `patients-stu3` directories respectively. I think it's fine if the directory structure appears as follows

image (with IPOP if it has people too)

**Review by mgramigna — APPROVED** [review]

File: EXM_104/Makefile [hunk 1]
```diff
@@ -1,47 +1,51 @@
...
 
+.setup-cqf-ruler-r4: .cqf-ruler connectathon .seed-cqf-ruler-r4
+	touch .setup-cqf-ruler-r4:
+
 .cqf-ruler:
 	docker run --name cqf-ruler --rm -dit -p 8080:8080 contentgroup/cqf-ruler
 	touch .cqf-ruler
 
 connectathon:
 	$(info connectathon checks out a specific commit SHA in case filepaths are updated)
```
Comment (lukoktonos) [inline EXM_104/Makefile]:
```suggestion
```

File: EXM_104/Makefile [hunk 2]
```diff
@@ -53,34 +57,70 @@ connectathon:
...
+	-d @connectathon/fhir4/bundles/EXM104_FHIR4-8.1.000/EXM104_FHIR4-8.1.000-files/library-EXM104_FHIR4-8.1.000.json
+	curl -X POST \
+  http://localhost:8080/cqf-ruler-r4/fhir \
+  -H 'Content-Type: application/json' \
+	-d @connectathon/fhir4/resources/library/all-libraries-bundle.json
+	touch .seed-cqf-ruler-no-vs-r4
+
 synthea:
-	git clone --single-branch --branch abacus --depth 1 https://github.com/projecttacoma/synthea.git
+	git clone --single-branch --branch exm_104_r4 --depth 1 https://github.com/projecttacoma/synthea.git
```
Comment (lukoktonos) [inline EXM_104/Makefile]:
```suggestion
	git clone --single-branch --branch abacus --depth 1 https://github.com/projecttacoma/synthea.git
```

Will need to be updated once https://github.com/projecttacoma/synthea/pull/10 gets in

File: EXM_104/Makefile [hunk 3]
```diff
@@ -53,34 +57,70 @@ connectathon:
...
+  -H 'Content-Type: application/json' \
+	-d @connectathon/fhir4/bundles/EXM104_FHIR4-8.1.000/EXM104_FHIR4-8.1.000-bundle.json
+	touch .seed-cqf-ruler-r4
+
+.seed-cqf-ruler-no-vs-r4:
+	until `curl --output /dev/null --silent --head --fail http://localhost:8080/cqf-ruler-r4`; do printf '.'; sleep 5; done
+	curl -X PUT \
+  http://localhost:8080/cqf-ruler-r4/fhir/Measure/measure-exm104-FHIR4 \
+  -H 'Content-Type: application/json' \
+	-d @connectathon/fhir4/resources/measure/measure-EX104_FHIR4-8.1.000.json
```
Comment (lukoktonos) [inline EXM_104/Makefile]:
```suggestion
	-d @connectathon/fhir4/bundles/EXM104_FHIR4-8.1.000/EXM104_FHIR4-8.1.000-files/measure-EXM104_FHIR4-8.1.000.json
```

File: EXM_104/Makefile [hunk 4]
```diff
@@ -53,34 +57,70 @@ connectathon:
...
+	curl -X POST \
+  http://localhost:8080/cqf-ruler-r4/fhir/ \
+  -H 'Content-Type: application/json' \
+	-d @connectathon/fhir4/bundles/EXM104_FHIR4-8.1.000/EXM104_FHIR4-8.1.000-files/library-EXM104_FHIR4-8.1.000.json
+	curl -X POST \
+  http://localhost:8080/cqf-ruler-r4/fhir \
+  -H 'Content-Type: application/json' \
+	-d @connectathon/fhir4/resources/library/all-libraries-bundle.json
```
Comment (lukoktonos) [inline EXM_104/Makefile]:
```suggestion
	-d @connectathon/fhir4/bundles/EXM105_FHIR4-8.1.000/EXM105_FHIR4-8.1.000-files/library-deps-EXM105_FHIR4-8.1.000-bundle.json
```

**Review by lukoktonos — CHANGES_REQUESTED** [review]
- [x] stu3 working with vs
- [x] stu3 working with no-vs (Library id needs to be `measure-EXM104-FHIR3-8.1.000` for PUT request)
- [x] r4 working with vs (~still getting calculation error mentioned in synthea pr both with cqf-ruler in docker and running locally on `develop` branch~ fixed with `mvn clean`).
- [x] r4 working with no-vs (paths need to be updated, added suggestions in-line)

**Comment by okeefm** [comment]
Fixed the seed issues/other suggested issuses. Should be ready for re-review.

File: EXM_104/Makefile [hunk 5]
```diff
@@ -1,86 +1,125 @@
...
+	-d @connectathon/fhir4/bundles/EXM104_FHIR4-8.1.000/EXM104_FHIR4-8.1.000-bundle.json
+	touch .seed-cqf-ruler-r4
+
+.seed-cqf-ruler-no-vs-r4:
+	until `curl --output /dev/null --silent --head --fail http://localhost:8080/cqf-ruler-r4`; do printf '.'; sleep 5; done
+	curl -X PUT \
+  http://localhost:8080/cqf-ruler-r4/fhir/Measure/measure-EXM104-FHIR4-8.1.000 \
+  -H 'Content-Type: application/json' \
+	-d @connectathon/fhir4/bundles/EXM104_FHIR4-8.1.000/EXM104_FHIR4-8.1.000-files/measure-EXM104_FHIR4-8.1.000.json
+	curl -X POST \
```
Comment (lukoktonos) [inline EXM_104/Makefile]:
```suggestion
	curl -X PUT \
```

File: EXM_104/Makefile [hunk 6]
```diff
@@ -1,86 +1,125 @@
...
+  http://localhost:8080/cqf-ruler-r4/fhir/ \
```
Comment (lukoktonos) [inline EXM_104/Makefile]:
```suggestion
  http://localhost:8080/cqf-ruler-r4/fhir/Library/library-EXM104-FHIR4-8.1.000 \
```

**Review by lukoktonos — CHANGES_REQUESTED** [review]

**Review by lukoktonos — APPROVED** [review]

**Pull request merged**